README.md

```diff
@@ -1,2 +1,555 @@
+  -h, --help  show this help message and exit
+```
+
+### Short name
+
+Usually, both flags and options come with a shortcut syntax.
+For example, instead of writing:
+```
+$ python test.py --some 1
+```
+
+One could write:
+```
+$ python test.py -s 1
+```
+
+To define a shortcut letter for a flag or an option, the `short` parameter of either `Flag` or `Opt` annotations should be used:
+```py
+# test.py
+
+from {pyname} import cli, Flag
+
+@cli
+def entry(some: Flag(short='s')):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+This allows invoking the CLI with the following syntax:
+```
+$ python test.py -s
+True
+```
+
+### Prefix
+
+Flags and options are usually called with the `-` prefix (in short and long variations).
+To override this behavior, the `prefix` parameter of either `Flag` or `Opt` annotations should be used.
+
+Consider the following example:
+```py
+# test.py
+
+from {pyname} import cli, Flag
+
+@cli
+def entry(some: Flag(prefix='+')):
+    print(some)
+
+if __name__ == '__main__':
+    cli()
+```
+
+Then the command line interface could be invoked as follows:
+```
+$ python test.py ++some
+True
+```
+
+### Types
+
+By default, the value that comes from the CLI, if it's an `Opt` or an `Arg`, is of `str` type.
+
+Consider the following example:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(some: Arg):
+    print(type(some))
+
+if __name__ == '__main__':
+    cli()
+```
+
+If an integer value is passed to the command line, the following output is produced:
+```
+$ python test.py 1
+<class 'str'>
+```
+
+To enforce the value to be an `int`, the following syntax should be used:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+@cli
+def entry(some: Arg[int]):
+    print(type(some))
+
+if __name__ == '__main__':
+    cli()
+```
+
+This makes `some` to be an integer:
+```
+$ python test.py 1
+<class 'int'>
+```
+
+But sometimes, a custom type should be parsed from a command line.
+This could be done by defining a parser for one using the `parse` decorator:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+class Custom:
+    ...
+
+@cli
+def entry(some: Arg[Custom]):
+    ...
+
+@cli.parse
+def custom(x: str) -> Custom:
+    ...
+
+if __name__ == '__main__':
+    cli()
+```
+
+Note: it's possible to manually add a parser for a type using the `add_parser` function.
+It works in the same way as the `parse` decorator:
+```py
+# test.py
+
+from {pyname} import cli, Arg
+
+class Custom:
+    def parse(x: str) -> 'Custom':
+        ...
+
+@cli
+def entry(some: Arg[Custom]):
+    ...
+
+cli.add_parser(Custom.parse)
+
+if __name__ == '__main__':
+    cli()
+```
+
+### Commands
+
+Complex command line interfaces like `git` have several subcommands: `git status`, `git pull`, `git push`, etc.
+These subcommands act as separate CLIs and, thus, are separate functions in {Name}.
+
+Consider the following example as a fake `git` CLI:
+```py
+# git.py
+
+from {pyname} import cli
+
+@cli
+def entry():
```

Comment:
I think that in this example `git` instead of `entry` would look much better.

Reply:
Fixed.